File: pythons/loop.py
while True :
    n = input ("Masukkan jumlah perulangan (ketik q untuk keluar):")
    if n == "q":
        break
    try:
        n = int(n)
    except ValueError:
        print("Input tidak valid. Harap masukkan angka.")
        continue
    
    for line in range (1, n+1):
        for number in range (1, line+1):
            print(number,end=" ")
        print()
    
    for line in range (n, 1, -1):
        for number in range (1, line): 
            print(number,end=" ")
        print()

File: javas/loop.java
import java.util.Scanner;

public class loop {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            System.out.print("Masukkan jumlah perulangan (ketik q untuk keluar):");
            String input = sc.nextLine();

            if (input.equals("q")) {
                break;
            }

            int n;
            try {
                n = Integer.parseInt(input);
            } catch (NumberFormatException e) {
                System.out.println("Input tidak valid. Harap masukkan angka.");
                continue;
            }

            for (int line = 1; line <= n; line++) {
                for (int number = 1; number <= line; number++) {
                    System.out.print(number + " ");
                }
                System.out.println();
            }

            for (int line = n; line > 1; line--) {
                for (int number = 1; number < line; number++) {
                    System.out.print(number + " ");
                }
                System.out.println();
            }
        }

        sc.close();
    }
}
